### PR TITLE
boards: toradex: add toradex verdin imx8mm board

### PR DESCRIPTION
**Pull request by hiagofranco**
Add Toradex Verdin iMX8M Mini board based on NXP MIMX8MM EVK using the i.MX8MM SoC. This code is intented to be used with the Cortex-M4.

Docs: https://builds.zephyrproject.io/zephyr/pr/61047/docs/boards/toradex/verdin_imx8mm/doc/index.html

### CODE REVIEW

**Comment by carlescufi** [comment]
@danieldegrasse could you please take a look?

File: boards/arm/verdin_imx8mm_m4/board.cmake
Comment (danieldegrasse) [inline boards/arm/verdin_imx8mm_m4/board.cmake:2]: Please update copyright

File: boards/arm/verdin_imx8mm_m4/Kconfig.defconfig
Comment (danieldegrasse) [inline boards/arm/verdin_imx8mm_m4/Kconfig.defconfig:11]: Why include this, if `CONFIG_XIP` is enabled by default in the board defconfig?

File: boards/arm/verdin_imx8mm_m4/verdin_imx8mm_m4_defconfig
Comment (danieldegrasse) [inline boards/arm/verdin_imx8mm_m4/verdin_imx8mm_m4_defconfig:13]: Is this Kconfig required as the "minimal boot set" of Kconfigs? IE does the UART driver need to be set to interrupt mode in order to function, or can the board boot without this Kconfig set

File: boards/arm/verdin_imx8mm_m4/board.cmake
Comment (hiagofranco) [reply boards/arm/verdin_imx8mm_m4/board.cmake]: Done, thanks

File: boards/arm/verdin_imx8mm_m4/Kconfig.defconfig
Comment (hiagofranco) [reply boards/arm/verdin_imx8mm_m4/Kconfig.defconfig]: Fixed, I actually based this one on the imx8mm evk from NXP. thanks

File: boards/arm/verdin_imx8mm_m4/verdin_imx8mm_m4_defconfig
Comment (hiagofranco) [reply boards/arm/verdin_imx8mm_m4/verdin_imx8mm_m4_defconfig]: Removed, thanks

**Comment by hiagofranco** [comment]
Hi @danieldegrasse, is there anything else I could do to help this PR moving forward? thanks

File: boards/arm/verdin_imx8mm_m4/verdin_imx8mm_m4.yaml
Comment (hiagofranco) [reply boards/arm/verdin_imx8mm_m4/verdin_imx8mm_m4.yaml:20]: > is ignore_tags needed to resolve a CI failure? If not, I'd like to remove it

I think so, I'd keep to prevent any errors in this regard.

> Can we add a supported block here? based on board documentation, I believe we need the following:

Nice, added, thanks!

**Review by danieldegrasse — DISMISSED** [review]

**Comment by hiagofranco** [comment]
Hi guys, sorry to tag you @mmahadevan108, @dleach02, and @decsny, but could you please review this PR so we can move forward? Thanks.

**Comment by decsny** [comment]
Should there be a toradex board group in the MAINTAINER.yml?

**Comment by hiagofranco** [comment]
> Should there be a toradex board group in the MAINTAINER.yml?

If that's required, I can update the PR. Please let me know.

**Comment by decsny** [comment]
> > Should there be a toradex board group in the MAINTAINER.yml?
> 
> If that's required, I can update the PR. Please let me know.

I think the project is trying to deprecate the CODEOWNERS file

**Comment by hiagofranco** [comment]
> > > Should there be a toradex board group in the MAINTAINER.yml?
> > 
> > 
> > If that's required, I can update the PR. Please let me know.
> 
> I think the project is trying to deprecate the CODEOWNERS file

Added, thanks! Please let me know if that's correct or if something else is needed.

**Comment by decsny** [comment]
@kartben is the maintainers entry correct, and does it require any approval by TSC since it adds a maintainer?

**Review by nordicjm — CHANGES_REQUESTED** [review]
Needs to be migrated to hwmv2

**Comment by hiagofranco** [comment]
> Needs to be migrated to hwmv2

Hello, apologies for the delay. I believe I was able to update the PR to hwmv2. Please let me know.

File: CODEOWNERS
Comment (danieldegrasse) [inline CODEOWNERS:104]: I don't think we should add to the codeowners file here, the rest of the file paths are out of date for boards and it seems like we may not have migrated them to HWMv2 (@nordicjm is there a reason we chose not to?) I know we are migrating to `MAINTAINERS.yml`, so perhaps that is why

File: boards/toradex/verdin_imx8mm/doc/img/verdin_imx8mm_m4.jpg
Comment (danieldegrasse) [inline boards/toradex/verdin_imx8mm/doc/img/verdin_imx8mm_m4.jpg:1]: I think we should keep filename of the board image generic, in case we add support for cores besides the CM4 target in the future.

File: boards/toradex/verdin_imx8mm/doc/index.rst
Comment (danieldegrasse) [inline boards/toradex/verdin_imx8mm/doc/index.rst:90]: File path here needs to be updated

File: boards/toradex/verdin_imx8mm/doc/index.rst
Comment (danieldegrasse) [inline boards/toradex/verdin_imx8mm/doc/index.rst:109]: File path needs updating

File: boards/toradex/verdin_imx8mm/doc/index.rst
Comment (danieldegrasse) [inline boards/toradex/verdin_imx8mm/doc/index.rst:132]: File path out of date

File: boards/toradex/verdin_imx8mm/doc/index.rst
Comment (danieldegrasse) [inline boards/toradex/verdin_imx8mm/doc/index.rst:166]: File path out of date, also please use `zephyr_file` here

File: boards/toradex/verdin_imx8mm/doc/index.rst
Comment (danieldegrasse) [inline boards/toradex/verdin_imx8mm/doc/index.rst:250]: Board target should be `verdin_imx8mm/mimx8mm6/m4`

File: CODEOWNERS
Comment (nordicjm) [reply CODEOWNERS:104]: Didn't notice that codeowners was out of data, will need to have a look at that. @nashif for comment on use of codeowners file

File: boards/toradex/verdin_imx8mm/doc/img/verdin_imx8mm_m4.jpg
Comment (nordicjm) [inline boards/toradex/verdin_imx8mm/doc/img/verdin_imx8mm_m4.jpg:1]: Change to webp then put through https://tinypng.com/

File: boards/toradex/verdin_imx8mm/verdin_imx8mm_mimx8mm6_m4.yaml
Comment (nordicjm) [inline boards/toradex/verdin_imx8mm/verdin_imx8mm_mimx8mm6_m4.yaml:12]: Does not make sense with what is in the description

**Comment by dleach02** [comment]
@hiagofranco do you plan on returning to your PR and addressing the comments?

**Comment by drewmoseley** [comment]
Hi @dleach02 I spoke to @hiagofranco and he has been moved to another role within Toradex which is why this has been stale for a while.  I'll try to take it up and move it forward and just consult with him directly.  We definitely want to see this support added in properly but I'm new to Zephyr so it may take some time.

**Comment by dleach02** [comment]
No problem. Ping the PR if you need assistance and to keep the stale label cleared.

File: CODEOWNERS
Comment (drewmoseley) [reply CODEOWNERS:104]: @nashif any update on this point? Should we remove the entry in the CODEOWNERS file?

**Comment by drewmoseley** [comment]
Please remove the stale label. This is still on my backlog.

**Comment by mmahadevan108** [comment]
@hiagofranco , any chance you plan to get back to this PR?

**Comment by drewmoseley** [comment]
> @hiagofranco , any chance you plan to get back to this PR?

It's now on my plate as @hiagofranco has been pulled into a different role.  I plan to get to but not sure when.

**Comment by drewmoseley** [comment]
Hi all. I finally got around to spending some time on this.  I'm not sure the best way to take over this PR since it is based on @hiagofranco's fork.  I [created a fork](https://github.com/drewmoseley/zephyr/tree/add_verdin_imx8mm_m4_board) and added additional commits to address most of (I hope) the feedback above.  For now I have kept them as separate commits for ease of review but am happy to squash them when we get closer to approval.

Should I just create a new PR with fork/branch and continue the discussion there?

**Comment by danieldegrasse** [comment]
> Should I just create a new PR with fork/branch and continue the discussion there?

Is it possible for you to push commits to his fork, if @hiagofranco sets up access permissions correctly? Otherwise, I do think we'd need to create an additional PR.

**Comment by kartben** [comment]
> > Should I just create a new PR with fork/branch and continue the discussion there?
> 
> Is it possible for you to push commits to his fork, if @hiagofranco sets up access permissions correctly? Otherwise, I do think we'd need to create an additional PR.

I was gonna suggest that too but then realized GitHub UI says "**Maintainers** are allowed to edit this pull request.", so not sure the branch would be writable by @drewmoseley -- but definitely worth trying!

**Comment by drewmoseley** [comment]
Confirmed that I cannot (yet) write to @hiagofranco's fork.  I've pinged him directly to see if he can share it with me.

**Comment by drewmoseley** [comment]
And just like that, I have push access.  Ready for further review.

File: CODEOWNERS
Comment (drewmoseley) [reply CODEOWNERS]: I have made these changes.

File: boards/toradex/verdin_imx8mm/doc/img/verdin_imx8mm_m4.jpg
Comment (drewmoseley) [reply boards/toradex/verdin_imx8mm/doc/img/verdin_imx8mm_m4.jpg:1]: This has been addressed in the latest push.

File: boards/toradex/verdin_imx8mm/doc/index.rst
Comment (drewmoseley) [reply boards/toradex/verdin_imx8mm/doc/index.rst]: This has been addressed in the latest push.

File: boards/toradex/verdin_imx8mm/doc/index.rst
Comment (drewmoseley) [reply boards/toradex/verdin_imx8mm/doc/index.rst]: This has been addressed in the latest push.

File: boards/toradex/verdin_imx8mm/doc/index.rst
Comment (drewmoseley) [reply boards/toradex/verdin_imx8mm/doc/index.rst]: This has been addressed in the latest push.

File: boards/toradex/verdin_imx8mm/doc/index.rst
Comment (drewmoseley) [reply boards/toradex/verdin_imx8mm/doc/index.rst]: This has been addressed in the latest push.

File: boards/toradex/verdin_imx8mm/doc/index.rst
Comment (drewmoseley) [reply boards/toradex/verdin_imx8mm/doc/index.rst]: This has been addressed in the latest push.

File: boards/toradex/verdin_imx8mm/doc/img/verdin_imx8mm_m4.jpg
Comment (drewmoseley) [reply boards/toradex/verdin_imx8mm/doc/img/verdin_imx8mm_m4.jpg:1]: This has been addressed in the latest push.

File: boards/toradex/verdin_imx8mm/verdin_imx8mm_mimx8mm6_m4.yaml
Comment (drewmoseley) [reply boards/toradex/verdin_imx8mm/verdin_imx8mm_mimx8mm6_m4.yaml:12]: Can you clarify? It's not clear to me what needs to change.  The RAM and FLASH on these boards is configurable so I don't really know what values should be listed here.

**Comment by mmahadevan108** [comment]
Please take a look at the Compliance failures

**Comment by drewmoseley** [comment]
Yes. At a minimum, looks like this needs a rebase.

File: boards/toradex/verdin_imx8mm/verdin_imx8mm_mimx8mm6_m4_defconfig
Comment (nordicjm) [inline boards/toradex/verdin_imx8mm/verdin_imx8mm_mimx8mm6_m4_defconfig:12]:
```suggestion
```

File: boards/toradex/verdin_imx8mm/doc/index.rst
Comment (nordicjm) [inline boards/toradex/verdin_imx8mm/doc/index.rst:68]: use proper board name as supplied to west build with \`\` around it and call it board target

File: boards/toradex/verdin_imx8mm/doc/img/verdin_imx8mm_m4.jpg
Comment (nordicjm) [reply boards/toradex/verdin_imx8mm/doc/img/verdin_imx8mm_m4.jpg:1]: Not resolved

File: MAINTAINERS.yml
Comment (nordicjm) [inline MAINTAINERS.yml:4186]: to be removed, none of these people are zephyr collaborators

**Review by nordicjm — CHANGES_REQUESTED** [review]
commits need to be squashed

File: MAINTAINERS.yml
Comment (drewmoseley) [reply MAINTAINERS.yml:4186]: I'm unclear; remove the whole addition to MAINTAINERS.yml or just the "collaborators" section?

**Comment by drewmoseley** [comment]
> commits need to be squashed

I will definitely do that once I get all the comments addressed.  Unless you prefer me to do it now; it's easier for me to keep track of what has been fixed and what is still open.

**Review by drewmoseley — COMMENTED** [review]
Based on [this commit](https://github.com/zephyrproject-rtos/zephyr/commit/d5a91a24de9255427213fa918f7643165f209c3d) I think this change is right but please confirm.  I'm a bit unsure as the "40-ohm" enum still [appears here](https://github.com/zephyrproject-rtos/zephyr/blob/4488ed1a20a97007d451d2ac1f093c413c56e77a/dts/bindings/pinctrl/nxp%2Cimx8m-pinctrl.yaml#L81).

File: MAINTAINERS.yml
Comment (nordicjm) [reply MAINTAINERS.yml]: the whole addition

**Comment by nordicjm** [comment]
> > commits need to be squashed
> 
> I will definitely do that once I get all the comments addressed. Unless you prefer me to do it now; it's easier for me to keep track of what has been fixed and what is still open.

I'm not going to review a PR which has fixups after fixups after fixups, it needs to be squashed now for people to review it

**Comment by drewmoseley** [comment]
OK.  Squashed and verified compliance test succeed locally.  Hopefully, I did not miss any review points.

File: boards/toradex/verdin_imx8mm/doc/index.rst
Comment (nordicjm) [inline boards/toradex/verdin_imx8mm/doc/index.rst:68]:
```suggestion
The Zephyr ``verdin_imx8mm/mimx8mm6/m4`` board target supports the following hardware
```

File: boards/toradex/verdin_imx8mm/verdin_imx8mm_mimx8mm6_m4.yaml
Comment (nordicjm) [inline boards/toradex/verdin_imx8mm/verdin_imx8mm_mimx8mm6_m4.yaml:21]:
```suggestion
vendor: toradex
```

**Review by nordicjm — COMMENTED** [review]
Minor changes, rest is ok

**Comment by drewmoseley** [comment]
Thanks @nordicjm your latest review comments have been included.

**Review by nordicjm — DISMISSED** [review]

**Comment by dleach02** [comment]
@hiagofranco you have a doc build problem with your boards doc page

**Comment by drewmoseley** [comment]
@dleach02 can you clarify?  I don't understand what and how to test.

File: boards/toradex/verdin_imx8mm/doc/index.rst
Comment (kartben) [inline boards/toradex/verdin_imx8mm/doc/index.rst:246]:
```suggestion
Here is an example for the :zephyr:code-sample:`hello_world` application.
```

This fixes the doc build failure

File: boards/toradex/verdin_imx8mm/doc/index.rst
Comment (kartben) [inline boards/toradex/verdin_imx8mm/doc/index.rst:12]: Image will be automatically inserted


```suggestion
```

File: boards/toradex/verdin_imx8mm/doc/index.rst
Comment (kartben) [inline boards/toradex/verdin_imx8mm/doc/index.rst:4]:
```suggestion
.. zephyr:board:: verdin_imx8mm
```

File: boards/toradex/verdin_imx8mm/board.yml
Comment (kartben) [inline boards/toradex/verdin_imx8mm/board.yml:2]:
```suggestion
  name: verdin_imx8mm
  full_name: Verdin iMX8M Mini
```

**Review by kartben — CHANGES_REQUESTED** [review]

**Comment by drewmoseley** [comment]
Thanks @kartben @dleach02.   Changes applied.

**Review by kartben — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Comment by kartben** [comment]
@hiagofranco and @holtmann many thanks for your patience and perseverance!

**Comment by drewmoseley** [comment]
This is excellent.  Thanks to all for the help and patience.  I know our marketing team will be happy to hear this.  Now we  need to consider which other of our modules we should add.

**Pull request merged**
